src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILInterpreter.cs

```diff
@@ -2461,10 +2462,19 @@ private struct MethodCallInfo
 
         private void InterpretCallDelegate(ref MethodCallInfo callInfo, ref LocalVariableSet localVariableSet)
         {
-            for (int i = callInfo.ArgCount; i > 0; i--)
+            for (int i = callInfo.LocalVariableTypes.Length - 1; i > 0; i--)
             {
                 StackItem stackItem = PopWithValidation();
-                TypeDesc argumentType = callInfo.Signature[i - 1];
+                TypeDesc argumentType = default;
+
+                if (i == 1 && !callInfo.IsStatic)
+                {
+                    argumentType = callInfo.OwningType;
```

Comment:
The type argument for instance value type methods is byref. Do we need to comment/TODO/assert about it?

Reply:
Added a `throw` and TODO further down in the code that handles the actual `call` opcode